pytorch_lightning/plugins/environments/cluster_environment.py

```diff
@@ -11,10 +11,12 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-from abc import ABC, abstractmethod
+from abc import abstractmethod
 
+from pytorch_lightning.plugins.base_plugin import Plugin
 
-class ClusterEnvironment(ABC):
+
+class ClusterEnvironment(Plugin):
```

Comment:
Hey! Great observation but I think we need to go a different path here. We have neglected so far to define a good base class for plugins that include cluster envs, because cluster envs don't really "do" that much. 

I would actually move what is in the Plugin class down to the training_type_plugin and annotate the signature with a Union of Training_typePlugin and ClusterEnvironment. 
We can then later refine the base class further :) What do you think?

Reply:
That's fine with me. @ananthsub your concern is related. What do you think about @awaelchli's suggestions

Reply:
Agreed with @awaelchli 's approach!